glomap/estimators/relpose_estimation.cc

```diff
@@ -3,69 +3,69 @@
+  std::vector<Eigen::Vector2d> points2D_1, points2D_2;
+  std::vector<char> inliers;
+
+  const size_t kNumChunks = 10;
+  size_t inverval = std::ceil(valid_pair_ids.size() / kNumChunks);
+  LOG(INFO) << "Estimating relative pose for " << valid_pair_ids.size()
             << " pairs";
-  for (image_pair_t chunks = 0; chunks < 10; chunks++) {
-    std::cout << "\r Estimating relative pose: " << chunks * 10 << "%"
+  for (size_t chunk_id = 0; chunk_id < kNumChunks; chunk_id++) {
+    std::cout << "\r Estimating relative pose: " << chunk_id * kNumChunks << "%"
               << std::flush;
-    image_pair_t start = chunks * inverval;
-    image_pair_t end = std::min((chunks + 1) * inverval, image_pair_ids.size());
-#pragma omp parallel for
-    for (image_pair_t pair = start; pair < end; pair++) {
-      ImagePair& image_pair = view_graph.image_pairs[image_pair_ids[pair]];
-      image_t idx1 = image_pair.image_id1;
-      image_t idx2 = image_pair.image_id2;
-
-      camera_t camera_id1 = images[idx1].camera_id;
-      camera_t camera_id2 = images[idx2].camera_id;
+    const size_t start = chunk_id * inverval;
+    const size_t end = std::min((chunk_id + 1) * inverval, valid_pair_ids.size());
 
+#pragma omp parallel for schedule(dynamic) private( \
+        points2D_1, points2D_2, inliers)
+    for (size_t pair_idx = start; pair_idx < end; pair_idx++) {
+      ImagePair& image_pair = view_graph.image_pairs[valid_pair_ids[pair_idx]];
+      const Image& image1 = images[image_pair.image_id1];
+      const Image& image2 = images[image_pair.image_id2];
       const Eigen::MatrixXi& matches = image_pair.matches;
 
       // Collect the original 2D points
-      std::vector<Eigen::Vector2d> points2D_1, points2D_2;
-      points2D_1.reserve(matches.rows());
-      points2D_2.reserve(matches.rows());
+      points2D_1.clear();
+      points2D_2.clear();
       for (size_t idx = 0; idx < matches.rows(); idx++) {
-        feature_t feature_id1 = matches(idx, 0);
-        feature_t feature_id2 = matches(idx, 1);
-
-        points2D_1.push_back(images[idx1].features[feature_id1]);
-        points2D_2.push_back(images[idx2].features[feature_id2]);
+        points2D_1.push_back(image1.features[matches(idx, 0)]);
+        points2D_2.push_back(image2.features[matches(idx, 1)]);
       }
 
-      // Estimate the relative pose with poselib
-      std::vector<char> inliers;
+      inliers.clear();
       poselib::CameraPose pose_rel_calc;
       poselib::estimate_relative_pose(
           points2D_1,
           points2D_2,
-          ColmapCameraToPoseLibCamera(cameras[images[idx1].camera_id]),
-          ColmapCameraToPoseLibCamera(cameras[images[idx2].camera_id]),
+          ColmapCameraToPoseLibCamera(cameras[image1.camera_id]),
+          ColmapCameraToPoseLibCamera(cameras[image2.camera_id]),
           options.ransac_options,
           options.bundle_options,
           &pose_rel_calc,
           &inliers);
```

Comment:
@lpanaf Is it intentional that we don't update the inliers of the image_pair here but keep the original colmap one?

Reply:
We calculate the inliers in later stage, with different two-view geometry: E, F, or H

Reply:
Seems to have some formatting problem here

Reply:
Thanks, I found it. Makes sense.